Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -43,23 +43,28 @@ message EventTypeQuery{
 // ie: ChooseWorkerCpu ---> PipExecutionStep_ChooseWorkerCpu
 
 enum ExecutionEventId{

```

Comment:
> ExecutionEventId{ [](start = 4, length = 18)

why do we need to this 'invalid' member? if it's a protobuf requirement, should it be added to all the enums? Also, I think we might want to have a helper function, so we do not forget that we need to +1 when converting from xlg -> proto

Reply:
It's not a protobuf requirement, its just since if our key contains {eventID = 0}, protobuf will serialize that as {} since it discards default values (0 is default value for int). This will cause problems for searching for the key. 

Also at no point do we convert xlg -> proto for this enum (the only place it is used is the key and where is no conversion being done here).

---
In reply to: [310239009](https://github.com/microsoft/BuildXL/pull/694#discussion_r310239009) [](ancestors = 310239009)

Reply:
Hmmm... I might be missing something here. But don't we need to have "Invalid = 0" on all of our enums?

---
In reply to: [310241033](https://github.com/microsoft/BuildXL/pull/694#discussion_r310241033) [](ancestors = 310241033,310239009)

Reply:
Yes, but if its not a key then it won't make a difference. The defacto standard in protobuf is that a missing or non-serialized field is the default value, so the consumer of the protoobject when seeing {} should handle it as being 0.

---
In reply to: [310244883](https://github.com/microsoft/BuildXL/pull/694#discussion_r310244883) [](ancestors = 310244883,310241033,310239009)